### ISSUE TITLE
New to TF and python: How to train the Generator/Discriminator for multiple iterations each epoch ?

### ISSUE THREAD

**Issue by Kdoing** [affinelayer/pix2pix-tensorflow#75]
Like the original implementation of the DCGAN, I am interested in adapting the code to train the Generator multiple times each epoch.
I am very new to TF and python, so I am having difficulties realizing this with the way the code is trained (using fetches).
I would greatly appreciate if anyone can give me some hints.

**Comment by dustyYMelody7** [comment]
I also want to know how to realize it? Have you solved it ? Could you give me some advice?

**Comment by julien2512** [comment]
What do you mean about DCGAN ? Is this the same than in
https://fr.slideshare.net/enakai/dcgan-how-does-it-work 
?

If it is the case, and you have a dataset with semantic parameters, there will be a lot to change as the code is set to learn image-2-image purposes.
You will need to completely remove the conv layer from the generator, and use your data on deconv inputs.

That is to say everywhere there is "input" or "conv" in the code :)

**Comment by davesean** [comment]
> I also want to know how to realize it? Have you solved it ? Could you give me some advice?

At the moment the model is trained by telling the session you want these:
"train=tf.group(update_losses, incr_global_step, gen_train, update_flip_prob)"
As a first step you would need to separate the training of the discriminator from the generator by removing this
"with tf.control_dependencies([discrim_train]):"
Now you need to control what is done in each iteration. You need to pass "discrim_train" and "gen_train" out of the scope. This can be done by adding two more strings to the Model collection at the top.
Then when you are in your training loop, you can decide if you want to only train the discriminator
>                fetches ={
>                   "train": model.discrim_train,
>                   "loss": model.update_losses
>                }
You'll also need to add "update_losses" to the Model and need to pass it if you want the losses to be updated.
This is how I think it would go, if anyone else wants to pitch in, feel free :)
